### ISSUE TITLE
Ultrabubble code finds a false ultrabubble

### ISSUE THREAD

**Issue by jeizenga** [vgteam/vg#672]
@glennhickey 
Encountered the problem on this graph: https://transfer.sh/sbG5F/primary-final-graph.int.bluntified.vg

`vg snarls  -l -t` identifies this Snarl:
```
{"type": 1, "start": {"node_id": 96339}, "end": {"node_id": 147584, "backward": true}}
```
However, this is not an ultrabubble. As you can see in this visualization, these traversals define a subgraph that is cyclic:
<img width="1090" alt="screen shot 2017-02-15 at 4 24 50 pm" src="https://cloud.githubusercontent.com/assets/7377501/23001893/544647e6-f39c-11e6-8071-85550f79b2aa.png">

**Comment by glennhickey** [comment]
Looking into it now.

**Comment by glennhickey** [comment]
Behaviour is inherited from vg::is_acyclic() fwiw

printf "96339 96340 96341 96342 96343 147584" > nodes.txt
vg index primary-final-graph.int.bluntified.vg -x
primary-final-graph.int.bluntified.xg
vg find -N nodes.txt -x primary-final-graph.int.bluntified.xg -c 1 >
subgraph.vg
vg stats -A subgraph.vg
acyclic

I have to finish my coffee, but does what you're showing fit our definition
of a bidirected cycle?  ie you can't walk around and around it with an
alternating path.

**Comment by jeizenga** [comment]
I hope not, but let's throw up some flares.
@adamnovak @benedictpaten 

My vote is no ultrabubble because I want to be sure that when a path enters one side of the bubble it leaves out the other side. If that's not true, it's going to mess up a lot of things I've developed. Bigly.

**Comment by jeizenga** [comment]
@adamnovak and I went back to the paper and I've changed my mind. I do think this is an ultrabubble but I'm not happy about it because it means I mean to change a lot of shit.

**Comment by glennhickey** [comment]
Talking to @benedictpaten, it's a bug after all.  The paper isn't super clear on the matter, but an ultrabubble is effectively a superbubbule -- so representable by a DAG.  This is, I believe, stricter than your original assumption, so your existing snarls code should be OK.  I'll revise the check in vg to use the new definition.

**Comment by glennhickey** [comment]
Closed by #677
